public/data.js

```diff
@@ -0,0 +1,53 @@
+class DataController {
+    static updateData(response, input) {
+        // Update data every time it changes in firestore
+        response.docChanges().forEach(change => {
+
+            const doc = { ...change.doc.data(), id: change.doc.id };
+
+            switch (change.type) {
+                case "added":
+                    input.push(doc);
+                    break;
+                case "modified":
+                    const index = input.findIndex(item => item.id == doc.id);
+                    input[index] = doc;
+                    break;
+                case "removed":
+                    input = input.filter(item => item.id !== doc.id);
+                    break;
+                default:
+                    break;
+            }
+        });
+    }
+
+    static getProject(update) {
```

Comment:
It wasn't immediately obvious that `update` was should be a callback. I don't think calling this getProject helped either. Consider `watchActiveProjects(onChange)`

Reply:
updated